### PR TITLE
Updating CAPI to Release Candidate Protocol

### PR DESCRIPTION
**Pull request by rbharath**
This PR updates CAPI to the latest release candidate version of the protocol

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -23,9 +23,11 @@ for a guide.
 
 
 You can install the core requirements with `pip` (make sure this is
-the `pip` in your virtual environment though)
+the `pip` in your virtual environment though).
```
Comment (robrobbins) [inline README.md]: don't much care for this sentence. more accurate would be to note that you should

1. have setup a virtualenv
2. have source .../bin/activated it

File: tests/cli/test_register.py [hunk 2]
```diff
@@ -1,9 +1,65 @@
...
+
+    # Approve the market token allowance
+    old_mkt_allowance = call(market_token.allowance(user, voting.address))
+    assert old_mkt_allowance == 0
+    tx = transact(market_token.approve(voting.address, w3.toWei(10, 'milliether'), opts={'from': user}))
+    rct = w3.eth.waitForTransactionReceipt(tx)
+    assert rct['status'] == 1
+    new_mkt_allowance = call(market_token.allowance(user, voting.address))
+    assert new_mkt_allowance == w3.toWei(10, 'milliether')
+    assert stake <= new_mkt_allowance
```
Comment (robrobbins) [inline tests/cli/test_register.py:68]: this is what i was talking about.

move all steps for setting up the register test to smaller tests that run before it. one very large test will be hard to refactor once we setup a helper or change the conftest.

you can be as piecemeal as you want with separate tests for `test_has_foo`, `test_can_bar` or a larger one that encompasses a logical outcome like `test_can_stake` but don't leave it all in the `register` test

File: tests/cli/test_resolution.py [hunk 3]
```diff
@@ -2,13 +2,65 @@
...
+
+    # Approve the market token allowance
+    old_mkt_allowance = call(market_token.allowance(user, voting.address))
+    assert old_mkt_allowance == 0
+    tx = transact(market_token.approve(voting.address, w3.toWei(10, 'milliether'), opts={'from': user}))
+    rct = w3.eth.waitForTransactionReceipt(tx)
+    assert rct['status'] == 1
+    new_mkt_allowance = call(market_token.allowance(user, voting.address))
+    assert new_mkt_allowance == w3.toWei(10, 'milliether')
+    assert stake <= new_mkt_allowance
```
Comment (robrobbins) [inline tests/cli/test_resolution.py:58]: same comment as above

File: tests/delivery/test_deliveries.py [hunk 4]
```diff
@@ -3,15 +3,71 @@
...
+    # Approve the market token allowance
+    old_mkt_allowance = call(market_token.allowance(user, voting.address))
+    assert old_mkt_allowance == 0
+    tx = transact(market_token.approve(voting.address, w3.toWei(10, 'milliether'), opts={'from': user}))
+    rct = w3.eth.waitForTransactionReceipt(tx)
+    assert rct['status'] == 1
+    new_mkt_allowance = call(market_token.allowance(user, voting.address))
+    assert new_mkt_allowance == w3.toWei(10, 'milliether')
+
+    assert stake <= new_mkt_allowance
```
Comment (robrobbins) [inline tests/delivery/test_deliveries.py:74]: same...

File: tests/listing/test_listings.py [hunk 5]
```diff
@@ -5,13 +5,69 @@
...
+
+    # Approve the market token allowance
+    old_mkt_allowance = call(market_token.allowance(user, voting.address))
+    assert old_mkt_allowance == 0
+    tx = transact(market_token.approve(voting.address, w3.toWei(10, 'milliether'), opts={'from': user}))
+    rct = w3.eth.waitForTransactionReceipt(tx)
+    assert rct['status'] == 1
+    new_mkt_allowance = call(market_token.allowance(user, voting.address))
+    assert new_mkt_allowance == w3.toWei(10, 'milliether')
+    assert stake <= new_mkt_allowance
```
Comment (robrobbins) [inline tests/listing/test_listings.py:75]: same

File: tests/task/test_task.py [hunk 6]
```diff
@@ -77,8 +77,63 @@ def test_get_non_existant_task(mock_get_task, test_client):
...
+
+    # Approve the market token allowance
+    old_mkt_allowance = call(market_token.allowance(user, voting.address))
+    assert old_mkt_allowance == 0
+    tx = transact(market_token.approve(voting.address, w3.toWei(10, 'milliether'), opts={'from': user}))
+    rct = w3.eth.waitForTransactionReceipt(tx)
+    assert rct['status'] == 1
+    new_mkt_allowance = call(market_token.allowance(user, voting.address))
+    assert new_mkt_allowance == w3.toWei(10, 'milliether')
+    assert stake <= new_mkt_allowance
```
Comment (robrobbins) [inline tests/task/test_task.py:140]: tra la la...

File: tests/voting/test_candidates.py [hunk 7]
```diff
@@ -29,13 +29,68 @@ def test_get_candidates_application(w3, voting, listing, test_client, dynamo_tab
...
+
+    # Approve the market token allowance
+    old_mkt_allowance = call(market_token.allowance(user, voting.address))
+    assert old_mkt_allowance == 0
+    tx = transact(market_token.approve(voting.address, w3.toWei(10, 'milliether'), opts={'from': user}))
+    rct = w3.eth.waitForTransactionReceipt(tx)
+    assert rct['status'] == 1
+    new_mkt_allowance = call(market_token.allowance(user, voting.address))
+    assert new_mkt_allowance == w3.toWei(10, 'milliether')
+    assert stake <= new_mkt_allowance
```
Comment (robrobbins) [inline tests/voting/test_candidates.py:94]: ...

**Review by robrobbins — CHANGES_REQUESTED** [review]
remember the part where i said i didnt mind repetition in tests for setups
but
it should be broken out into either small tests for individual pieces
or
larger tests that encompassed one logical operation

this is that

File: README.md (hunk 1 above)
Comment (rbharath) [reply README.md]: Good suggestion, expanded out README with virtualenv directions

File: tests/cli/test_register.py (hunk 2 above)
Comment (rbharath) [reply tests/cli/test_register.py:68]: Factored out into smaller tests

File: tests/cli/test_resolution.py (hunk 3 above)
Comment (rbharath) [reply tests/cli/test_resolution.py:58]: Done

**Comment by rbharath** [comment]
I've factored out the state into different tests as suggested. The fact these tests are repeated in multiple places makes me feel that they should be fixtures instead, but that should be an easy refactor now that the tests are cleanly separated. This should be ready for a second round of review

**Comment by robrobbins** [comment]
right. as we discussed already a helper or a fixture, whichever fits better

**Review by robrobbins — APPROVED** [review]

**Pull request merged**